microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoWorker.java

```diff
@@ -0,0 +1,299 @@
+package de.gsi.microservice.concepts.majordomo;
+
+import static de.gsi.microservice.concepts.majordomo.MajordomoProtocol.MdpMessage;
+import static de.gsi.microservice.concepts.majordomo.MajordomoProtocol.MdpWorkerCommand;
+import static de.gsi.microservice.concepts.majordomo.MajordomoProtocol.MdpWorkerMessage;
+import static de.gsi.microservice.concepts.majordomo.MajordomoProtocol.receiveMdpMessage;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.management.ManagementFactory;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zeromq.SocketType;
+import org.zeromq.ZContext;
+import org.zeromq.ZMQ;
+
+import de.gsi.microservice.rbac.RbacRole;
+import de.gsi.microservice.utils.SystemProperties;
+
+/**
+ * Majordomo Protocol Client API, Java version Implements the MajordomoProtocol/Worker spec at
+ * http://rfc.zeromq.org/spec:7.
+ *
+ * default heart-beat time-out [ms] is set by system property: 'OpenCMW.heartBeat' // default: 2500 [ms]
+ * default heart-beat liveness is set by system property: 'OpenCMW.heartBeatLiveness' // [counts] 3-5 is reasonable
+ * N.B. heartbeat expires when last heartbeat message is more than HEARTBEAT_INTERVAL * HEARTBEAT_LIVENESS ms ago.
+ * this implies also, that worker must either return their message within 'HEARTBEAT_INTERVAL * HEARTBEAT_LIVENESS ms' or decouple their secondary handler interface into another thread.
+ *
+ */
+public class MajordomoWorker extends Thread {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MajordomoWorker.class);
+    private static final int HEARTBEAT_LIVENESS = SystemProperties.getValueIgnoreCase("OpenCMW.heartBeatLiveness", 3); // [counts] 3-5 is reasonable
+    private static final int HEARTBEAT_INTERVAL = SystemProperties.getValueIgnoreCase("OpenCMW.heartBeat", 2500); // [ms]
+    private static final AtomicInteger WORKER_COUNTER = new AtomicInteger();
+
+    // ---------------------------------------------------------------------
+    private static final int BLOCKING_QUEUE_DEFAULT_SIZE = 10;
+    protected final String uniqueID;
+    protected final ZContext ctx;
+    private final String brokerAddress;
+    private final String serviceName;
+    private final byte[] serviceBytes;
+
+    private final AtomicBoolean run = new AtomicBoolean(true);
+    private final SortedSet<RbacRole<?>> rbacRoles;
```

Comment:
Codacy found an issue: [Perhaps 'rbacRoles' could be replaced by a local variable.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)